File: src/main/java/com/derongan/minecraft/mineinabyss/Ascension/Effect/Effects/AscensionEffect.java
package com.derongan.minecraft.mineinabyss.Ascension.Effect.Effects;

import org.bukkit.entity.Player;

public interface AscensionEffect {
    void applyEffect(Player player, int ticks);
    boolean isDone();
    int getRemainingTicks();
    void setRemainingTicks(int remainingTicks);
    void cleanUp(Player player);
}

File: mineinabyss-components/src/main/kotlin/com/mineinabyss/components/curse/AscensionEffect.kt
package com.mineinabyss.components.curse

import org.bukkit.entity.Player
import kotlin.time.Duration

interface AscensionEffect {
    fun applyEffect(player: Player, ticks: Int)
    val isDone: Boolean
    val duration: Duration
    fun cleanUp(player: Player)

    fun clone(): AscensionEffect
}
